kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/balancer/impl/RandomBalancerStrategy.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.client.balancer.impl;
+
+import java.util.List;
+
+public class RandomBalancerStrategy extends AbstractBalancerStrategy {
+
+    private List<String> availableEndpoints;
+
+    public RandomBalancerStrategy(List<String> availableEndpoints) {
+        this.availableEndpoints = availableEndpoints;
+    }
+
+    @Override
+    public String next() {
+        checkEmpty(availableEndpoints);
+
+        if (availableEndpoints.size() == 1) {
+            return availableEndpoints.get(0);
+        }
+
+        int index = getRandomInt(0, availableEndpoints.size()-1);
+
+        return availableEndpoints.get(index);
+    }
+
+    @Override
+    public void markAsOffline(String url) {
+        synchronized (availableEndpoints) {
+            availableEndpoints.remove(url);
+        }
+    }
+
+    @Override
+    public void markAsOnline(String url) {
+        synchronized (availableEndpoints) {
+            availableEndpoints.add(url);
+        }
+    }
+
+    protected int getRandomInt(int min, int max) {
+        return (int) Math.floor(Math.random() * (max - min + 1)) + min;
```

Comment:
sure we could do this but this strategy is more of a demo one rather than a real production ready strategy. So not sure there is need for performance improvements here.